sympy/sets/sets.py

```diff
@@ -165,7 +340,37 @@ def _union(self, other):
 
         Used within the :class:`Union` class
         """
-        return None
+        from sympy.polys.polyerrors import PolynomialError
+        from sympy.sets.fancysets import ImageSet
+
+        def _check_imageset(self):
+            # Returns True if all the args are ImageSet
+            # otherwise false
+            if isinstance(self, ImageSet):
+                return True
+            elif isinstance(self, Union):
+                for self_img in self.args:
+                    if not isinstance(self_img, ImageSet):
+                        return False
+                return True
+            else:
+                return False
+        try:
+            if _check_imageset(self) and _check_imageset(other):
+                return self._union_simplify(other)
+            else:
+                return None
+        except (
+            NotImplementedError,
+            PolynomialError
+        ):
+            # `_union_simplify` is for simplifying trigonometric eq. solution
```

Comment:
I shifted the code in `fancysets/ImageSet._union` . Now It works for all other cases.